internal/typeinfo/member.go

```diff
@@ -79,32 +102,43 @@ type structField struct {
 	omitEmpty bool
 }
 
-// OuterType returns the reflected type of struct in
-// which this Member implementation is a field.
-func (f *structField) OuterType() reflect.Type {
+// ArgType returns the type of struct in this field is located in.
+func (f *structField) ArgType() reflect.Type {
 	return f.structType
 }
 
-// MemberName returns the name of this struct field.
-func (f *structField) MemberName() string {
-	return f.tag
+// LocateParams locates the struct the field is located in from the typeToValue
+// map. It returns the value of this field. A slice with a single entry is
+// returned to fit the Input interface.
+func (f *structField) LocateParams(typeToValue map[reflect.Type]reflect.Value) ([]reflect.Value, error) {
+	s, err := locateValue(typeToValue, f.structType)
+	if err != nil {
+		return nil, err
+	}
+	return []reflect.Value{s.Field(f.index)}, nil
 }
 
-// ValueFromOuter returns the value of this field in the input reflected struct.
-func (f *structField) ValueFromOuter(v reflect.Value) (reflect.Value, error) {
-	return v.Field(f.index), nil
+// String returns a natural language description of the struct field for use in
+// error messages.
+func (f *structField) String() string {
+	return "tag \"" + f.tag + "\" of struct \"" + f.structType.Name() + "\""
 }
 
-// GetScanTarget returns a pointer for the target of rows.Scan, and a ScanProxy
-// reference in the event that we need to coerce that pointer into a struct
-// field.
-// Rows.Scan will return an error if it tries to scan NULL into a type that
+// LocateScanTarget locates the struct specified in structField from the
+// provided typeToValue map. It returns a pointer for the target of rows.Scan,
+// and a ScanProxy reference in the event that we need to coerce that pointer
+// into a struct field.
+// rows.Scan will return an error if it tries to scan NULL into a type that
 // cannot be set to nil, so for types that are not a pointer and do not
 // implement sql.Scanner, a pointer to them is generated and passed to
 // Rows.Scan. If Scan has set this pointer to nil the value is zeroed by
 // ScanProxy.OnSuccess.
```

Comment:
I think this comment is really necessary when you read the code. I am wondering if we should move it to the `LocateScanTarget` in the interface itself instead. What do you think?